src/python/botan3.py

```diff
@@ -1825,3 +1831,88 @@ def srp6_client_agree(username, password, group, hsh, salt, b, rng):
                                                                     rng.handle_(),
                                                                     a, al,
                                                                     k, kl))
+
+def zfec_encode(k, n, input_bytes):
+    """
+    ZFEC-encode an input message according to the given parameters
+
+    :param int k: the number of shares required to recover the original
+    :param int n: the total number of shares
+    :param bytes input_bytes: the input message, in bytes
+
+    :returns: n arrays of bytes, each one containing a single share
+    """
+    input_size = len(input_bytes)
+    # note: the C++ code checks that input_bytes is a multiple of k
+    outsize = input_size // k
+    p_p_nbytes = c_char_p * n
+
+    # allocate memory for the outputs (create_string_buffer makes bytes)
+    outputs = [
+        create_string_buffer(outsize)
+        for a in range(n)
+    ]
+
+    c_outputs = p_p_nbytes(*[
+        addressof(output)
+        for output in outputs
+    ])
+
+    # actual C call
+    x = _DLL.botan_zfec_encode(
+        c_size_t(k), c_size_t(n), input_bytes, c_size_t(input_size), c_outputs
+    )
+    if x != 0:
+        raise RuntimeError(
+            "Unexpected error, code={}".format(x)
+        )
```

Comment:
This check on the return value should not be required, because `ffi_api` sets the `errcheck` property to a function (`_errcheck`) which already checks that the return value is 0 and throws otherwise.

Reply:
Fixed in d251adf1b